Lib/ctypes/__init__.py

```diff
@@ -265,7 +265,47 @@ class c_void_p(_SimpleCData):
 class c_bool(_SimpleCData):
     _type_ = "?"
 
-from _ctypes import POINTER, pointer, _pointer_type_cache
+def POINTER(cls):
+    """Create and return a new ctypes pointer type.
+
+    Pointer types are cached and reused internally,
+    so calling this function repeatedly is cheap.
+
+    Pointer types for incomplete types are not cached,
+    so calling this function repeatedly will give
+    different types.
+    """
+    if cls is None:
+        return c_void_p
+    try:
+        pt = cls.__pointer_type__
+        if pt is not None:
+            return pt
+    except AttributeError:
+        pass
+    if isinstance(cls, str):
+        return type(f'LP_{cls}', (_Pointer,), {})
+    return type(f'LP_{cls.__name__}', (_Pointer,), {'_type_': cls})
+
+def pointer(obj):
+    """Create a new pointer instance, pointing to 'obj'.
+
+    The returned object is of the type POINTER(type(obj)). Note that if you
+    just want to pass a pointer to an object to a foreign function call, you
+    should use byref(obj) which is much faster.
+    """
+    typ = POINTER(type(obj))
+    return typ(obj)
+
+def set_non_ctypes_pointer_type(cls, pointer_type):
+    """Set pointer type for cls."""
+    assert not isinstance(pointer_type, _Pointer)
+    assert not isinstance(cls, _SimpleCData)
+
+    cls.__pointer_type__ = pointer_type
```

Comment:
Please revert the helper.  This would be acceptable in a test file, but does not look helpful.  The assertions are incorrect (use `issubclass` or pass metaclass), and ctype types' `__pointer_type__` is read only.